src/TahoeLAFS/Storage/API.hs

```diff
@@ -257,13 +267,32 @@ instance FromHttpApiData ByteRanges where
 instance ToHttpApiData ByteRanges where
     toHeader = renderByteRanges
 
+data LeaseSecret = Renew ByteString | Cancel ByteString
+
+instance FromHttpApiData LeaseSecret where
+    parseHeader bs =
+        do
+            let [key, val] = BS.split 32 bs
+            case key of
+                "lease-renew-secret" -> bimap pack Renew $ Base64.decode val
+                _ -> bimap pack Cancel $ Base64.decode val
```

Comment:
Probably worth having better error handling here (ie, do something better for the string "blahblah-secret").

This can be addressed in a follow-up.

Reply:
I don't see this code in the repo after the merge, it's gone!